auth: unused lambda capture warnings in LUA records code

<!-- Tell us what is issue is about -->
 - Program: Authoritative
 - Issue type: Bug report/Feature request <!-- delete the one that does not apply -->

### Short description
clang warns about unused variables in LUA record lambda's.

### Environment
<!-- Tell us about the environment -->
 - Operating system: Arch
 - Software version: git (master as of d26856a32e7506817c291c49d8eb5a60129dd751)
 - Software source: git

### Steps to reproduce
```
autoreconf -fi
CXX=clang++ CC=clang LD=lld ./configure --without-modules --without-dynmodules --enable-lua-records
make
```

### Expected behaviour
No warnings.

### Actual behaviour
clang 8 gives this warning:

```
lua-record.cc:537:40: warning: lambda capture 'bestwho' is not used [-Wunused-lambda-capture]
  lua.writeFunction("createReverse", [&bestwho,&query,&zone](string suffix, boost::optional<std::unordered_map<string,string>> e){
                                      ~^~~~~~~~
lua-record.cc:537:56: warning: lambda capture 'zone' is not used [-Wunused-lambda-capture]
  lua.writeFunction("createReverse", [&bestwho,&query,&zone](string suffix, boost::optional<std::unordered_map<string,string>> e){
                                                     ~~^~~~
lua-record.cc:619:41: warning: lambda capture 'bestwho' is not used [-Wunused-lambda-capture]
  lua.writeFunction("createReverse6", [&bestwho,&query,&zone](string suffix, boost::optional<std::unordered_map<string,string>> e){
                                       ~^~~~~~~~
lua-record.cc:619:57: warning: lambda capture 'zone' is not used [-Wunused-lambda-capture]
  lua.writeFunction("createReverse6", [&bestwho,&query,&zone](string suffix, boost::optional<std::unordered_map<string,string>> e){
                                                      ~~^~~~
lua-record.cc:806:33: warning: lambda capture 'counter' is not used [-Wunused-lambda-capture]
  lua.writeFunction("report", [&counter](string event, boost::optional<string> line){
                     
```

--- a/pdns/lua-record.cc
+++ b/pdns/lua-record.cc
@@ -534,7 +534,7 @@ std::vector<shared_ptr<DNSRecordContent>> luaSynth(const std::string& code, cons
     });
 
 
-  lua.writeFunction("createReverse", [&bestwho,&query,&zone](string suffix, boost::optional<std::unordered_map<string,string>> e){
+  lua.writeFunction("createReverse", [&query](string suffix, boost::optional<std::unordered_map<string,string>> e){
       try {
       auto labels= query.getRawLabels();
       if(labels.size()<4)
@@ -616,7 +616,7 @@ std::vector<shared_ptr<DNSRecordContent>> luaSynth(const std::string& code, cons
     });
 
 
-  lua.writeFunction("createReverse6", [&bestwho,&query,&zone](string suffix, boost::optional<std::unordered_map<string,string>> e){
+  lua.writeFunction("createReverse6", [&query](string suffix, boost::optional<std::unordered_map<string,string>> e){
       vector<ComboAddress> candidates;
 
       try {
@@ -802,8 +802,7 @@ std::vector<shared_ptr<DNSRecordContent>> luaSynth(const std::string& code, cons
     });
 
 
-  int counter=0;
-  lua.writeFunction("report", [&counter](string event, boost::optional<string> line){
+  lua.writeFunction("report", [](string event, boost::optional<string> line){
       throw std::runtime_error("Script took too long");
     });
   if (g_luaRecordExecLimit > 0) {

